lib/pbench/test/unit/agent/test_utils.py

```diff
@@ -216,12 +218,42 @@ class TestLocalRemoteHost:
     """
 
     @staticmethod
-    def test_methods():
+    def test_methods(monkeypatch):
+        monkeypatch.setattr(
+            "ifaddr.get_adapters",
+            lambda: [
+                ifaddr.Adapter(
+                    "lo",
+                    "lo",
+                    [
+                        ifaddr.IP("127.0.0.1", 8, "lo"),
+                        ifaddr.IP(("::1", 0, 0), 128, "lo"),
+                    ],
+                    1,
+                ),
+                ifaddr.Adapter(
+                    "eth0",
+                    "eth0",
+                    [
+                        ifaddr.IP("10.1.1.1", 255, "eth0"),
+                        ifaddr.IP(("2600::1", 0, 0), 64, "eth0"),
+                        ifaddr.IP(("fe80::1", 0, 2), 64, "eth0"),
+                    ],
+                    2,
+                ),
+            ],
+        )
         lrh = LocalRemoteHost()
-        assert not lrh.is_local(
-            "testhost.example.com"
-        ), "'testhost.example.com' (which is not a real name) should still be remote"
+
+        # An unresolvable host isn't usable for benchmarking; we expect name
+        # resolution to fail, and the exception should propagate to the caller.
+        with pytest.raises(socket.gaierror):
+            lrh.is_local("testhost.example.com")
         assert not lrh.is_local("example.com"), "'example.com' should be remote"
```

Comment:
Why does `testhost.example.com` cause an exception to be raised...or, rather, why _doesn't_ the same happen for `example.com`??

Reply:
Because `example.com` really exists, but `test.example.com` doesn't. I stopped handling the error internally because I really don't think we want to. We're going to fail at some point if they specify a bad remote; why fudge here?

```
dbutenho 16:26 pbench-fio:is-remote ping example.com
PING example.com (93.184.216.34) 56(84) bytes of data.
64 bytes from 93.184.216.34 (93.184.216.34): icmp_seq=1 ttl=57 time=28.4 ms

dbutenho 17:18 pbench-fio:is-remote ping test.example.com
ping: test.example.com: Name or service not known
```

Reply:
I concur with and support your choice.  I just didn't realize that `example.com` actually existed.